core/src/test/java/org/elasticsearch/indices/memory/breaker/CircuitBreakerServiceIT.java

```diff
@@ -409,6 +407,24 @@ public void testCustomCircuitBreakerRegistration() throws Exception {
         assertThat(breaks, greaterThanOrEqualTo(1));
     }
 
+    public void testCanResetUnreasonableSettings() {
+        if (noopBreakerUsed()) {
+            logger.info("--> noop breakers used, skipping test");
+            return;
+        }
+        Settings insane = Settings.builder()
+            .put(IN_FLIGHT_REQUESTS_CIRCUIT_BREAKER_LIMIT_SETTING.getKey(), "5b")
+            .build();
+        client().admin().cluster().prepareUpdateSettings().setTransientSettings(insane).get();
+
+        reset();
```

Comment:
Maybe add a comment to this line that this'd throw an exception if you can't reset unreasonable settings.